Parallelize test running

Drake-Eidukas opened the pull request:
This PR does 2 things:
- Installs `pytest-xdist` and `pytest-sugar` in the docker image (requires running `docker-compose build honeybadgermpc`)
- Sets the tests to run with how many cpu "cores" your system has (i.e. 4 core + hyperthreading => 8 worker threads) as the parallelism factor.

On my system, this cut out about a minute of test runtime.

Drake-Eidukas commented:
BTW, let me know if I should be adding other people as reviewers, or just you two all the time.

smkuls commented on pytest.ini:
```diff
@@ -2,6 +2,10 @@
 log_file_level = DEBUG
+
+# Sets the parallelism of the tests that are curretnly running.
+addopts = -nauto
```
How many cores does this use?

Drake-Eidukas replied:
It uses a default number that makes sense based on your hardware, basically number of cores (x2 if you have hyperthreading)

smkuls replied:
Cool

smkuls commented:
@Drake-Eidukas You can also add @rahulgovind as a reviewer but I think only Andrew and I have the permissions to approve the code.

smkuls commented:
You should try to make sure that your commit message does not go on to two lines. Just like you have a very nice PR description, I'd like you to keep a detailed commit message too. This helps us to know what changes went in when looking at the logs.

Drake-Eidukas commented:
Alright, I'll add more detail to the message. I'll update the comment too.

Drake-Eidukas commented:
@smkuls Should be good now

smkuls reviewed (approved):
LGTM!

smkuls commented:
@Drake-Eidukas can you rebase this with the `dev` branch once? I dont want to do it from the UI since it does a merge and not a rebase which adds an extra merge commit. I will merge this after you rebase.

Drake-Eidukas committed:
Add pytest-xdist and pytest-sugar dependencies, enable test parallelism

This commit adds the pytest-xdist and pytest-sugar dependencies to our `setup.py`
file (requires rebuilding the docker image), as well as sets the parallelism to
default to the number of "cores" on your system (x2 if you have hyperthreading).

Drake-Eidukas commented:
@smkuls it's rebased
